agent/app/agent_capability.go

```diff
@@ -61,29 +64,45 @@ const (
 	capabilityEFS                               = "efs"
 	capabilityEFSAuth                           = "efsAuth"
 	capabilityEnvFilesS3                        = "env-files.s3"
+	capabilityExec                              = "execute-command"
+	capabilityDepsRootDir                       = "/managed-agents"
+	capabilityExecBinRelativePath               = "bin"
+	capabilityExecConfigRelativePath            = "config"
+	capabilityExecCertsRelativePath             = "certs"
 )
 
-var nameOnlyAttributes = []string{
-	// ecs agent version 1.19.0 supports private registry authentication using
-	// aws secrets manager
-	capabilityPrivateRegistryAuthASM,
-	// ecs agent version 1.22.0 supports ecs secrets integrating with aws systems manager
-	capabilitySecretEnvSSM,
-	// ecs agent version 1.27.0 supports ecs secrets for logging drivers
-	capabilitySecretLogDriverSSM,
-	// support ecr endpoint override
-	capabilityECREndpoint,
-	// ecs agent version 1.23.0 supports ecs secrets integrating with aws secrets manager
-	capabilitySecretEnvASM,
-	// ecs agent version 1.27.0 supports ecs secrets for logging drivers
-	capabilitySecretLogDriverASM,
-	// support container ordering in agent
-	capabilityContainerOrdering,
-	// support full task sync
-	capabilityFullTaskSync,
-	// ecs agent version 1.39.0 supports bulk loading env vars through environmentFiles in S3
-	capabilityEnvFilesS3,
-}
+var (
+	nameOnlyAttributes = []string{
+		// ecs agent version 1.19.0 supports private registry authentication using
+		// aws secrets manager
+		capabilityPrivateRegistryAuthASM,
+		// ecs agent version 1.22.0 supports ecs secrets integrating with aws systems manager
+		capabilitySecretEnvSSM,
+		// ecs agent version 1.27.0 supports ecs secrets for logging drivers
+		capabilitySecretLogDriverSSM,
+		// support ecr endpoint override
+		capabilityECREndpoint,
+		// ecs agent version 1.23.0 supports ecs secrets integrating with aws secrets manager
+		capabilitySecretEnvASM,
+		// ecs agent version 1.27.0 supports ecs secrets for logging drivers
+		capabilitySecretLogDriverASM,
+		// support container ordering in agent
+		capabilityContainerOrdering,
+		// support full task sync
+		capabilityFullTaskSync,
+		// ecs agent version 1.39.0 supports bulk loading env vars through environmentFiles in S3
+		capabilityEnvFilesS3,
+	}
+	capabilityExecRequiredBinaries = []string{
+		"amazon-ssm-agent",
+		"ssm-session-worker",
```

Comment:
I think you will also need to check for ssm-agent-worker

Reply:
nice catch! thanks for pointing it out